EventDataStores/ManyTablesPostgreSQLDataStore/ManyTablesPostgreSQLDatastore.cs

```diff
@@ -0,0 +1,269 @@
+        _updateConnection = updateConn;
+        
+        Console.WriteLine($"Started PostgresSQL database with script {Path.GetFileName(_initScriptPath)}");
+#if DEBUG
+        Console.WriteLine($"PostgresSQL ConnectionString: {_container.GetConnectionString()}");
+#endif
+    }
+
+    public async Task ResetAsync()
+    {
+        if (_insertConnection is not null)
+        {
+            await _insertConnection.CloseAsync();
+            await _insertConnection.DisposeAsync().AsTask();
+        }
+
+        if (_updateConnection is not null)
+        {
+            await _updateConnection.CloseAsync();
+            await _updateConnection.DisposeAsync().AsTask();
+        }
+
+        if (_container is not null)
+        {
+            await _container.DisposeAsync().AsTask();
+        }
+
+        await StartAsync();
+    }
+
+    public async Task AddOrUpdateFlightAsync(Flight flight)
+    {
+        // need to also create the icao tables if they do not exist. Store this information in memory or use "create table if not exists"
+        foreach (var airport in flight.GetAllAirports().Distinct())
+        {
+            if (_tableSet.Contains(airport)) continue;
+
+            string createTableSql = 
+                $"""
+                 CREATE TABLE {airport} (
+                     flightIdentification VARCHAR(36) UNIQUE PRIMARY KEY NOT NULL,
+                     isRecalculating BOOL NOT NULL DEFAULT (FALSE),
+                     lastWeather INT,
+                     departure TIMESTAMP NOT NULL,
+                     arrival TIMESTAMP NOT NULL
+                 );
+                 """;
+            var createTableCmd = new NpgsqlCommand(createTableSql, _insertConnection);
+            await createTableCmd.ExecuteNonQueryAsync().ConfigureAwait(false);
+
+            await using (var batch = new NpgsqlBatch(_insertConnection))
+            {
+                string createIndexSql1 = 
+                    $"""
+                     CREATE INDEX {airport}_events_idx
+                     ON {airport} (lastWeather, isRecalculating, departure, arrival, flightIdentification);
+                     """;
+                var batchCmd1 = new NpgsqlBatchCommand(createIndexSql1);
+                batch.BatchCommands.Add(batchCmd1);
+
+                string createIndexSql2 = 
+                    $"""
+                     CREATE INDEX {airport}_recalc_idx
+                     ON {airport} (flightIdentification, isRecalculating) WHERE isRecalculating = FALSE;
+                     """;
+                var batchCmd2 = new NpgsqlBatchCommand(createIndexSql2);
+                batch.BatchCommands.Add(batchCmd2);
+
+                string createIndexSql3 = 
+                    $"""
+                     CREATE INDEX {airport}_update_idx
+                     ON {airport} (flightIdentification, lastWeather);
+                     """;
+                var batchCmd3 = new NpgsqlBatchCommand(createIndexSql3);

```

Comment:
I think this can be removed if we want to reduce the footprint of this data store. I don't think we query based on `(flightIdentification, lastWeather)` anywhere.